docs/design/simple-rolling-update.md

```diff
@@ -0,0 +1,37 @@
+## Simple rolling update
+This is a lightweight design document for simple rolling update in ```kubectl```
+
+### Lightweight rollout
+Assume that we have a current replication controller named ```foo``` and it is running image ```image:v1```
+
+```kubectl rollingupdate rc foo --image=myimage:v2```
+
+Under the hood, this will do the following:
+   * Create a new replication controller named foo-next, identical to foo, except image is updated to the new image.
+   * Do the standard down/up one-by-one until foo has replicas zero and foo-next has replicas N
+   * delete rc foo
+   * create rc foo that is identical to foo-next
+   * delete foo-next
```

Comment:
@bgrant0607 
I would really prefer not to leave a user in a situation where their pods are unmanaged, so my preference would be to update the validation rules as @jlowdermilk suggests.

I'd much rather have 2 replication controllers temporarily controlling the same pods, than temporarily have no replication controllers controlling the pods.

Especially if we add validation so that you really can't resize them.

@davidopp 
Added some details on annotations.